client/extensions/wp-super-cache/wrap-settings-form.jsx

```diff
@@ -50,16 +93,22 @@ const wrapSettingsForm = getFormSettings => SettingsForm => {
 				handleToggle: this.handleToggle,
 			};
 
-			return <SettingsForm { ...this.props } { ...utils } />;
+			return (
+				<div>
+					<QuerySettings siteId={ this.props.siteId } />
+					<SettingsForm { ...this.props } { ...utils } />
+				</div>
+			);
 		}
 	}
 
 	const connectComponent = connect(
-		() => {
-			// TODO: Replace with REST API once ready.
-			const settings = {
+		state => {
+			const siteId = getSelectedSiteId( state );
+			const settings = Object.assign( {}, getSettings( state, siteId ), {
 				// Easy
-				scrules: true,
+				// http_only setting is not persisted, but is needed for the "Test Cache" action.
+				http_only: true,
```

Comment:
When you say "settings object", do you mean the value returned by the`getSettings` function call? If so, that result is the response from the API endpoint, which could return `true` for `http_only` as well, but it's not a value that is stored by WPSC so it's just as easy to set the initial value client-side. It should be the only value that is set in this way. All others will eventually come from the API. But perhaps I've misunderstood the question?